sqlx-postgres/src/types/hstore.rs

```diff
@@ -0,0 +1,221 @@
+use std::{
+    collections::{btree_map, BTreeMap},
+    mem::size_of,
+    ops::{Deref, DerefMut},
+    str::from_utf8,
+};
+
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    decode::Decode,
+    encode::{Encode, IsNull},
+    error::BoxDynError,
+    types::Type,
+    PgArgumentBuffer, PgTypeInfo, PgValueRef, Postgres,
+};
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Deserialize, Serialize)]
```

Comment:
It'd be nice to have a short documentation blurb and maybe an example or two.

If you'd like to take a shot at writing it, be my guest, otherwise I'll add some myself when I have the time.

Reply:
No worries, basic documentation added. Give me a shout if I missed anything.